src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -83,6 +83,9 @@ private void WriteWithUnseekableDataStream(TarEntryFormat format, Stream destina
             // We know the exact location where the data starts depending on the format
             long dataStartPosition = headerStartPosition + dataLocation;
 
+            // Before writing, update the offset field now that the entry belongs to an archive
+            _dataOffset = dataStartPosition + 1;
```

Comment:
Ok this is a bug: I should not do a `+1` otherwise if you position the archive stream here, and try to read the next byte, you will end up reading the 2nd byte from the data, not the first byte.

Fixing it now.

Reply:
Fixed https://github.com/dotnet/runtime/pull/105007/commits/6a03ccadbb65fa45aac25ea104406e5814b41de0